rotkehlchen/chain/avalanche/manager.py

```diff
@@ -0,0 +1,229 @@
+            self,
+            accounts: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumEthAddress, FVal]:
+        """Returns a dict with keys being accounts and balances in AVAX
+
+        May raise:
+        - RemoteError if an external service such as Covalent is queried and
+          there is a problem with its query.
+        """
+        balances = {}
+        for account in accounts:
+            balances[account] = self.get_avax_balance(account)
+        return balances
+
+    def get_block_by_number(self, num: int) -> Dict[str, Any]:
+        """Returns the block object corresponding to the given block number
+
+        May raise:
+        - RemoteError if an external service such as Covalent is queried and
+        there is a problem with its query.
+        """
+
+        block_data: MutableAttributeDict = MutableAttributeDict(self.w3.eth.get_block(num))  # type: ignore # pylint: disable=no-member  # noqa: E501
+        block_data['hash'] = hex_or_bytes_to_str(block_data['hash'])
+        return dict(block_data)
+
+    def get_code(self, account: ChecksumEthAddress) -> str:
+        """Gets the deployment bytecode at the given address
+
+        May raise:
+        - RemoteError if Covalent is used and there is a problem querying it or
+        parsing its response
+        """
+
+        return hex_or_bytes_to_str(self.w3.eth.getCode(account))
+
+    def get_transaction_receipt(
+            self,
+            tx_hash: str,
+    ) -> Dict[str, Any]:
+        tx_receipt = self.covalent.get_transaction_receipt(tx_hash)
+        if tx_receipt is False:
+            tx_receipt = self.w3.eth.get_transaction(tx_hash).__dict__  # type: ignore
+            return tx_receipt
+        try:
+            # Turn hex numbers to int
+            tx_receipt.pop('from_address_label', None)
+            tx_receipt.pop('to_address_label', None)
+            block_number = tx_receipt['block_height']
+            tx_receipt['blockNumber'] = tx_receipt.pop('block_height', None)
+            tx_receipt['cumulativeGasUsed'] = tx_receipt.pop('gas_spent', None)
+            tx_receipt['gasUsed'] = tx_receipt['cumulativeGasUsed']
+            successful = tx_receipt.pop('successful', None)
+            tx_receipt['status'] = 1 if successful else 0
+            tx_receipt['transactionIndex'] = 0
+            # TODO input and nonce is decoded in Covalent api, encoded in future
+            tx_receipt['input'] = '0x'
+            tx_receipt['nonce'] = 0
+            for index, receipt_log in enumerate(tx_receipt['log_events']):
+                receipt_log['blockNumber'] = block_number
+                receipt_log['logIndex'] = receipt_log.pop('log_offset', None)
+                receipt_log['transactionIndex'] = 0
+                tx_receipt['log_events'][index] = receipt_log
+        except (DeserializationError, ValueError) as e:
+            raise RemoteError(
+                f'Couldnt deserialize transaction receipt data from covalent {tx_receipt}',
```

Comment:
can you also add the reason in the error?
```suggestion
                f'Couldnt deserialize transaction receipt data from covalent {tx_receipt}' due to {str(e)},
```